source/agora/common/ManagedDatabase.d

```diff
@@ -0,0 +1,69 @@
+/*******************************************************************************
+
+    Contains a managed wrapper around SQLite databases.
+
+    This class ensures that all D2SQLite databases handled by the thread
+    are cleanly destroyed after the thread shuts down.
+
+    Copyright:
+        Copyright (c) 2020 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.common.ManagedDatabase;
+
+import d2sqlite3.database;
+import d2sqlite3.sqlite3;
+
+/// Ditto
+public class ManagedDatabase
+{
+    /// ManagedDatabase managed by the current thread
+    private static Database*[] thread_dbs;
+
+    /// Close all database handles
+    static ~this ()
+    {
+        foreach (db; thread_dbs)
+            db.close();
```

Comment:
If a SqliteException occurs here, how will it be handled?

Reply:
The application will exit with an error code and a stack trace will be emitted to the console.

It might not be a bad idea to log this. I'll update.

Reply:
But in practice there's nothing much more we can do other than to log the failure.